packages/tradershub/src/components/Base/Clipboard/Clipboard.tsx

```diff
@@ -0,0 +1,55 @@
+import React, { useEffect, useRef, useState } from 'react';
+import { useCopyToClipboard, useHover } from 'usehooks-ts';
+import { Button, useBreakpoint } from '@deriv/quill-design';
+import ClipboardIcon from '../../../public/images/clipboard.svg';
+import CheckmarkCircle from '../../../public/images/ic-checkmark-circle.svg';
+import { Tooltip } from '../Tooltip';
+
+type TClipboardProps = {
+    textCopy: string;
+};
+
+/**
+ * Props for the Clipboard component.
+ * @property {string} textCopy - The text to be copied to the clipboard.
+ *
+ * @example
+ * ```jsx
+ * <Clipboard textCopy="Text to be copied" />
+ * ```
+ */
+const Clipboard = ({ textCopy }: TClipboardProps) => {
+    const [, copy] = useCopyToClipboard();
+    const { isMobile } = useBreakpoint();
+    const [isCopied, setIsCopied] = useState(false);
+    const hoverRef = useRef(null);
+    const isHovered = useHover(hoverRef);
+    let timeoutClipboard: ReturnType<typeof setTimeout>;
+
+    /**
+     * Handle click event to copy text to clipboard.
+     * @param {React.MouseEvent<HTMLButtonElement, MouseEvent>} event - The click event.
+     */
+    const onClick = (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
+        setIsCopied(true);
+        copy(textCopy);
+        timeoutClipboard = setTimeout(() => {
+            setIsCopied(false);
+        }, 2000);
```

Comment:
```suggestion
        timeoutClipboard = setTimeout(() => {
            setIsCopied(false);
        }, 2000);
```
Instead of handling the transition here, u can use css to fade in the checkmark